RFC-0004-sparse-fill-value/README.md

```diff
@@ -0,0 +1,613 @@
+                                               values=[[0.2492, 0.2503], [0.3044, 0.3057]],
+                                               size=A.size(),
+                                               fill_value=[0.2232, 0.2220])
+       ```
+
+       The alternative, if the fill value was defined as scalar,
+       would be memory inefficient because the result of
+       `A.softmax(0)` is a full sparse tensor with all elements
+       specified.
+
+6.  The fill value specified that is specified in the `fill_value`
+    argument of the sparse tensor constructors may have different
+    (smaller) shape than the fill value of the hybrid tensor (as
+    defined by point 5).
+
+    The specified fill value (after converting it to `torch.Tensor`
+    instance) can be acquired via `_fill_value()` method.
+
+    For example, the fill value of a (1+1)-D hybrid tensor can be
+    specified as scalar:
+
+    ```python
+    A = torch.sparse_coo_tensor(indices=[[0, 3]],
+                                values=[[.11, .12], [.31, .32]],
+                                size=(4, 2),
+                                fill_value = 1.2)
+    A.fill_value() -> torch.tensor([1.2, 1.2])
+    A._fill_value() -> torch.tensor(1.2)
+    ```
+
+    The output of `fill_value()` is computed as
+
+    ```python
+    A._fill_value().resize(A.values().shape[1:])
+    ```
+
+    Storing the specified fill value instead of the fill value of the
+    hybrid tensor has several advantages:
+
+    - lower memory consumption,
+    - optimal evaluation of element-wise functions, see point 8 below,
+    - optimal detection of zero fill value, see point 10 below.
+
+
+7.  The fill value of a sparse tensor can be changed in place.
+
+    For instance:
+
+    ```python
+    A._fill_value().fill_(1.2)
+    ```
+
+    resets the fill value of a sparse tensor `A` to be `1.2`.
+
+8.  If `A` is a sparse tensor and `f` is any calculus function that is
+    applied to a tensor element-wise, then:
+
+    ```python
+    f(A) == torch.sparse_coo_tensor(A.indices(), f(A.values()), fill_value=f(A._fill_value()))
+    ```
+
+    Note that if `A` would be using COO storage format then this
+    relation holds only if `A` is coalesced (`A.values()` would throw
+    an exception otherwise).
+
+    From this relation follows an identity:
+
+    ```python
+    f(A).to_dense() == f(A.to_dense())
+    ```
+
+    that will be advantageous in testing the sparse tensor support of
+    element-wise functions.
+
+9.  The fill value of an element-wise n-ary operation on sparse
+    tensors with different fill values is equal to the result of
+    applying the operation to the fill values of the sparse
+    tensors.
+
+    For instance (`*` represents unspecified element),
+
+    ```
+    A = [[1, *], [3, 4]]                  # A fill value is 2
+    B = [[*, 6], [*, 8]]                  # B fill value is 7
+    A + B = [[1 + 3, 2 + 4], [*, 6 + 8]]  # A + B fill value is 2 + 7 = 9
```

Comment:
```suggestion
    A = [[1, *], [*, 4]]                  # A fill value is 2
    B = [[*, *], [7, 8]]                  # B fill value is 6
    A + B = [[1 + 6, *], [2 + 7, 4 + 8]]  # A + B fill value is 2 + 7 = 9
```

Reply:
This issue is now fixed with a modified example.